lib/karaokium/performances/performance.ex

```diff
@@ -24,4 +26,70 @@ defmodule Karaokium.Performances.Performance do
     |> cast(attrs, [:karaoke_id, :team_id, :song_id, :voting?])
     |> validate_required([:karaoke_id, :team_id, :song_id, :voting?])
   end
+
+  @doc """
+  Calculates a performance's score.
+
+  Uses [Wikipedia's first method](https://en.wikipedia.org/wiki/Quartile#Method_1) to calculate quartile values.
+  From these, determines the outliers and excludes such values from the final average calculation.
+
+  Preloads votes if they are not already loaded.
+  Returns 0 if there are no votes.
+
+  ## Examples
+
+      iex> votes = [%Karaokium.Polling.Vote{pontuation: 1} | List.duplicate(%Karaokium.Polling.Vote{pontuation: 9}, 5)]
+      iex> performance = %Karaokium.Performances.Performance{votes: votes}
+      iex> Karaokium.Performances.Performance.score(performance)
+      #Decimal<9.00>
+
+      iex> performance = %Karaokium.Performances.Performance{votes: []}
+      iex> Karaokium.Performances.Performance.score(performance))
+      #Decimal<0>
+
+      iex> performance = %Karaokium.Performances.Performance{}
+      iex> Karaokium.Performances.Performance.score(performance))
+      #Decimal<0>
+
+  """
+  def score(%__MODULE__{votes: %Ecto.Association.NotLoaded{}} = performance) do
+    if Ecto.assoc_loaded?(performance.votes) do
+      performance
+    else
+      Repo.preload(performance, :votes)
+    end
```

Comment:
@nelsonmestevao Could this if statement be omitted as the function declaration specifies that votes are not loaded in  `votes: %Ecto.Association.NotLoaded{}`

Reply:
You are totally right! :clap: